python/cudf/cudf/core/dataframe.py

```diff
@@ -157,42 +160,90 @@ class DataFrame(object):
     """
 
     def __init__(self, data=None, index=None, columns=None, dtype=None):
-        keys = index
-        if index is None:
-            index = RangeIndex(start=0)
-        self._index = as_index(index)
-        self._size = len(index)
-        self._cols = OrderedDict()
-        # has initializer?
-
-        if data is not None:
-            if is_list_like(data) and len(data) > 0:
+        super().__init__()
 
-                if isinstance(data[0], tuple):
-                    index = self._index = RangeIndex(start=0, stop=len(data))
-                    data = enumerate(itertools.zip_longest(*data))
+        if isinstance(columns, cudf.MultiIndex):
+            self.multi_cols = columns
+            columns = RangeIndex(len(columns))
 
-                elif not isinstance(data[0], list):
-                    # a nested list is something pandas supports and
-                    # we don't support list-like values in a record yet
-                    self._add_rows(data, index, keys)
-                    return
+        if data is None:
+            if index is None:
+                self._index = RangeIndex(0)
+            else:
+                self._index = as_index(index)
+            if columns is not None:
+                self._data = OrderedDict.fromkeys(
+                    columns,
+                    column.column_empty(
+                        len(self), dtype="object", masked=True
```

Comment:
Do we want to default to `string` columns here? I know Pandas uses `object` here but that's more general for them than Strings.